# EDA Toy Project - Starbucks & Ediya
___

## 1. 문제 정의

아래 주장의 사실 여부를 파악하고자 한다

***"이디야는 스타벅스의 임점 위치를 따라 개점하는, 이른바 '스타벅스 옆자리 전략'으로 무임승차한다."***

### 세부 단계  
- 1.1. 자료 선정  

- 1.2. 자료 수집 및 요약

- 1.3. 결론

### 1.1. 문제 상황 관련 자료 선정

해당 문제는 2015년부터 현재까지 언급되는 주장이다   
**스타벅스 이디야**를 검색창에 입력하면 관련 게시물도 나온다.

<img src="./img/001_kwsearch.png" width="35%" height="35%">

주장이 참임을 가정하고 작성된 게시물과 그 반대의 경우를 확인할 수 있다.  

목표 설정을 위해 다음을 기준으로 자료를 수집·요약해 기존 주장을 확인하고자 한다.
1. 작성자와 작성 시기가 명확한 글 (뉴스기사 or 일관성 있게 작성된 개인 블로그) - **신뢰**할 수 있는 글
2. 주장과 근거가 있는 글 (수치에 기반하지 않더라도 차용) - **논리**가 있는 글
3. 입점 전략에 관한 글 (입점이 언급되지 않은 마케팅 전략은 제외) - **관련성**이 높은 글

### 1.2. 관련 자료 수집 및 요약

<details>
<summary><strong>References</strong></summary>
<p>

1. 참 관련 자료
- https://www.speconomy.com/news/articleView.html?idxno=316421  
(스페셜경제, "이디야커피, 스타벅스 따라가면 성공한다(?), 2022년)  
- https://www.edaily.co.kr/news/read?newsId=01105366609365640   
(이데일리, "이디야의 필승 전략..'스타벅스 옆을 사수하라'", 2015년)  
- http://www.iconsumer.or.kr/news/articleView.html?idxno=10133  
(소비자평가, "‘스타벅스를 따라다닌 이디야’, 이디야 문창기 대표가 펼친 특별한 전략", 2019년)  
- https://m.blog.naver.com/PostView.naver?isHttpsRedirect=true&blogId=qaz030800&logNo=221680589189  
(네이버 블로그[작성자: 윈에이드], "유명기업의 마케팅전략: '이디야(EDIYA)커피' 편, 2019년)  
- https://m.blog.naver.com/urimmoe/222027146026  
(네이버 블로그[작성자: 로하스김], "커피전문점 이디야의 스타벅스 미투마케팅 전략", 2020년)  
- https://www.topdaily.kr/articles/90068  
 (톱데일리, "이디야커피 성장 비결, 스타벅스 따라다니기?", 2022년)  
<br>

2. 거짓 관련 자료  
- https://www.seoul.co.kr/news/newsView.php?id=20160919004005  
(서울신문, "스타벅스 vs 이디야, 매장 위치에 전략이 숨어 있다", 2016년)  
- https://brunch.co.kr/@tipoon1004/66  
(브런치[작성자: 권오은], "이디야는 스타벅스 옆을 떠난 걸까", 2017년)  
- https://www.insight.co.kr/news/194890  
(인사이트, "왜 스타벅스 옆에는 항상 토종 커피 브랜드 '이디야'가 있을까?", 2018년)  
- https://pikle.co/column/article/29709/  
(분당일보, "이디야가 스타벅스 옆에만 있다는 소문에 화나서 분석한 자료", 2021년) 
</p>
</details>


#### 1.2.1 참 관련 자료 요약 (총 6건)  
- 작성자 생활권에서 육안 관찰한 내용을 바탕으로 이디야의 입점 전략을 '미투 전략<sup>1</sup>'의 일환으로 설명
    - '실제로 ... 발견하는 것이 어렵지 않다' (2015년)  
    - '실제로 ... 떠올리거나 살펴보시기 바랍니다' (2020년)  
    - ' 지금도 스타벅스 매장 주변을 보면 ..." (2022년)  
<br>
- 관계자 언급  

    >  "브랜드 론칭 초기에는 ...(중략)... 스타벅스 옆을 노리는 전략을 펼쳤다" (2015년)

#### 1.2.2. 거짓 관련 자료 요약 (총 4건)
- 각 브랜드의 공개된 데이터(직접 수집 혹은 공공데이터)를 분석  
- 입점 전략 비교와 함꼐 관련 용어가 소개됨  
    - 스타벅스의 '허브 앤 스포크 전략<sup>2</sup>'  
    - 이디야의 '포인트 투 포인트 전략<sup>3</sup>'  
<br>
- 관계자 언급  

    > "사실 스타벅스는 옆은 임대료가 비싸서 ...(중략)... 이디야는 입점하기 힘들다" (2018년)

### 1.3. 결론
참 관련 자료는 대체로 주장을 가정하고 작성되었으므로, 입증하는 자료로 보기는 어렵다.  
거짓 관련 자료는 데이터를 제시해 주장이 100% 참이 아닐 수 있음을 제시했다.  

하지만 2015년과 2018년 각각 다른 입장에 힘을 보태는 이디야 관계자의 언급,  
'실제로 이디야가 스타벅스 근처에 자주 보인다'는 무시할 수만은 없는 기사 작성자의 직관,  
기존 분석된 방법의 타당성 등을 면밀히 살펴 분석 목표를 좁혀야 한다.  
___

## 2. 데이터 수집

### 2.1. 모듈 Import

In [34]:
# dataframe 핸들링을 위한 numpy, pandas 모듈
import numpy as np
import pandas as pd

# 동적 페이지를 다루기 위한 selenium 모듈
from selenium import webdriver
from selenium.common.exceptions import WebDriverException
from selenium.webdriver.chrome.service import Service as ChromeService
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support.wait import WebDriverWait
import selenium.webdriver.support.expected_conditions as EC

# 데이터 추출을 위한 beautifulsoup 모듈
from bs4 import BeautifulSoup
from urllib.request import Request

# 기타 작업에 유용한 모듈
from tqdm import tqdm_notebook
import time


### 2.2. 스타벅스 웹 데이터 수집

In [35]:
# webdriver 실행 및 페이지 접근
options = webdriver.ChromeOptions()
service = ChromeService(executable_path="chromedriver.exe")
driver = webdriver.Chrome(service=service, options=options)
driver.get('https://www.starbucks.co.kr/store/store_map.do')

In [38]:
driver.refresh()

In [42]:
# [지역 검색] 클릭
def click_local():
    xpath_local = '//*[@id="container"]/div/form/fieldset/div/section/article[1]/article/header[2]/h3/a'
    WebDriverWait(driver, 10).until(
        EC.presence_of_element_located((By.XPATH, xpath_local))).click()


# [서울] 클릭
def click_seoul():
    xpath_seoul = '//*[@id="container"]/div/form/fieldset/div/section/article[1]/article/article[2]/div[1]/div[2]/ul/li[1]/a'
    WebDriverWait(driver, 10).until(
        EC.presence_of_element_located((By.XPATH, xpath_seoul))).click()


# [전체] 클릭
def click_entire():
    xpath_entire = '//*[@id="mCSB_2_container"]/ul/li[1]/a'
    WebDriverWait(driver, 10).until(
        EC.presence_of_element_located((By.XPATH, xpath_entire))).click()



In [43]:
# 각 단계를 거쳐 서울 전체 가맹점 화면에 접근
click_local()
click_seoul()
click_entire()


In [ ]:
# html source로 soup 객체 출력
html = driver.page_source
soup = BeautifulSoup(html)


In [45]:
# 매장 정보가 담긴 container 
cont_branch = soup.find_all('li', class_='quickResultLstCon')
len(cont_branch)


610

In [ ]:
lat = 'data-lat'
lng = 'data-long'
addr = []

In [49]:
branch.find('p', class_='result_details').text

'서울특별시 중랑구 봉화산로 35 1522-3232'

In [48]:
for branch in cont_branch:
    try:
        print(branch['data-name'])
    except:
        print('no name')

동대문두타
동대문디자인플라자
국립중앙의료원
동묘앞역
동대문공원
대학로
신당역
주얼리시티
종로플레이스
신당역사거리
역삼아레나빌딩
논현역사거리
신사역성일빌딩
국기원사거리
대치재경빌딩R
봉은사역
압구정윤성빌딩
코엑스별마당
삼성역섬유센터R
압구정R
수서역R
양재강남빌딩R
선릉동신빌딩R
봉은사로선정릉
강남오거리
스타필드코엑스몰R
강남구청정문
도곡공원
강남R
대치은마사거리
청담영동대로
압구정
신사가로수
청담스타R
강남우성
학동사거리
대치사거리
포스코사거리
을지병원사거리
신사역
도곡역
서울세관사거리
청담사거리
한티역
역삼초교사거리
코엑스몰
가로수길
역삼대로
역삼포스코
강남대로
청담
삼성도심공항
삼성교
학동역
대치
SSG마켓도곡R
삼성
선릉세화빌딩
압구정역
압구정로
구역삼사거리
큰길타워
강남구청역
대치역
포이
아크플레이스
학여울역
선정릉역
삼성현대힐
GS타워
압구정미소
선릉역
코엑스사거리
역삼럭키
신압구정
아크로힐스논현
강남비젼타워
압구정로데오역
르네상스사거리
포스코
선릉로
논현힐탑
도산가로수길
대치한티
수서역사거리
코엑스아셈타워
센터필드R
도산사거리
강남세곡
대치삼성프라자
언주역
강남역신분당역사
삼성역
테헤란로아남타워
강남논현
삼성동
역삼이마트
미아뉴타운
강북구청사거리
미아사거리역
수유역
미아역
파라스파라서울
발산역사거리
마곡역
우장산역
마곡사이언스타워R
방화DT
마곡나루역
마곡GMG
가양역
화곡DT
양천향교역
화곡동
등촌
염창역
발산역
까치산역
화곡역8번출구
강서우장산
등촌대로
가양역사거리
내발산
마곡웰튼
김포공항DT
강서NC
마곡나루공원
강서구청사거리
낙성대역
관악서울대입구R
신림DT
서울대역
낙성대DT
사당
보라매대교
신림사거리
신림
신림타임스트림
서울대입구역
봉천역
건국대
어린이대공원역
아차산역
올림픽대교북단
구의DT
자양사거리
군자역
광나루
건대스타시티
건대입구
구의역
광장
건국클래식
건대후문
중곡역
자양이마트
뚝섬유원지역
군자사거리
자양
금천구청역
가산에스케이
독산사거리
금천시흥
가산디지털단지역
금천독산
가산디지털
W-Mall
마리오아울렛
가산브이타워
가산그레이

In [4]:
# 결과물 DataFrame 틀 만들기
def GenerateDataFrame(Area = [], Branch = [], Addr = []):
    tmp = (
        {'Area': Area,
        'Branch' : Branch,
        'Address' : Addr,
        }
    )
    return pd.DataFrame(tmp)

starbucks = GenerateDataFrame([], [], [])
starbucks.head()

Empty DataFrame
Columns: [Area, Branch, Address]
Index: []

In [5]:
# 구 내부에서 정보 가져오는 함수

def getInfo_gu(Area):
    branch = []; addr = []
    info = driver.find_elements(By.CLASS_NAME, 'quickResultLstCon')
    for i in info:
        ActionChains(driver).move_to_element(i).perform()

        li = i.text.split('\n')[:2]
        branch.append(li[0])
        addr.append(li[1])
        
    return [branch, addr]

In [6]:
# Initial empty DataFrame: "starbucks"
starbucks = GenerateDataFrame([], [], [])

# Approaching with area number: 25 area starting from [2] '강남구' to [26] '중랑구'
for n in tqdm_notebook(range(2,27)):
    gu = driver.find_element(By.XPATH, f'//*[@id="mCSB_2_container"]/ul/li[{n}]/a')
    if n >= 23:
        ActionChains(driver).move_to_element(gu).perform()
    gu_name = gu.text # 클릭 전 구 이름 변수 지정
    gu.click() # 구 클릭
    time.sleep(1)
    
    info = getInfo_gu(gu_name)
    
    starbucks_tmp = GenerateDataFrame([gu_name for i in range(len(info[0]))], info[0], info[1])
    starbucks = pd.concat([starbucks, starbucks_tmp])
    
    # Click the button "[지역검색]"
    LocaClick()
    time.sleep(1)

    # Click the button "[서울]" 
    SeoulClick()
    time.sleep(1)

# 데이터 확인
starbucks.head()

C:\Users\kjaeh\AppData\Local\Temp\ipykernel_11328\3136862288.py:5: TqdmDeprecationWarning: This function will be removed in tqdm==5.0.0
Please use `tqdm.notebook.tqdm` instead of `tqdm.tqdm_notebook`
  for n in tqdm_notebook(range(2,27)):


  0%|          | 0/25 [00:00<?, ?it/s]

Area    Branch                     Address
0  강남구     역삼이마트     서울특별시 강남구 역삼로 310 (역삼동)
1  강남구       삼성동  서울특별시 강남구 영동대로96길 12 (삼성동)
2  강남구      강남논현    서울특별시 강남구 강남대로 512 (논현동)
3  강남구  테헤란로아남타워    서울특별시 강남구 테헤란로 311 (역삼동)
4  강남구       삼성역    서울특별시 강남구 테헤란로 620 (대치동)

In [7]:
# Checking the DataFrame

starbucks.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 593 entries, 0 to 7
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Area     593 non-null    object
 1   Branch   593 non-null    object
 2   Address  593 non-null    object
dtypes: object(3)
memory usage: 18.5+ KB


In [10]:
# Resetting the index
starbucks.reset_index(inplace=True)

# Remove previous index
del starbucks['index']

# Reset index again, and set as main index
starbucks.reset_index(inplace=True)
starbucks.set_index('index')
starbucks.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 593 entries, 0 to 592
Data columns (total 5 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   level_0  593 non-null    int64 
 1   index    593 non-null    int64 
 2   Area     593 non-null    object
 3   Branch   593 non-null    object
 4   Address  593 non-null    object
dtypes: int64(2), object(3)
memory usage: 23.3+ KB


In [11]:
# 결측치 확인
print(len(starbucks['Area'].isna())) # 구 정보 결측치 개수
print(len(starbucks['Branch'].isna())) # 지점 정보 결측치 개수
print(len(starbucks['Address'].isna())) # 주소 결측치 개수

593
593
593


In [12]:
# 구 정보 확인
gu_result = starbucks['Area'].unique()
print(gu_result, f'\nLength: {len(gu_result)}')

['강남구' '강동구' '강북구' '강서구' '관악구' '광진구' '구로구' '금천구' '노원구' '도봉구' '동대문구' '동작구'
 '마포구' '서대문구' '서초구' '성동구' '성북구' '송파구' '양천구' '영등포구' '용산구' '은평구' '종로구' '중구'
 '중랑구'] 
Length: 25


In [ ]:
# starbucks.to_csv('data/01_starbucks.csv', sep= ',', encoding = 'cp949')
driver.close()

## 문제 2. 이디야 매장 정보 DataFrame으로 불러오기


In [14]:
driver = webdriver.Chrome('chromedriver.exe')
driver.get('https://www.ediya.com/contents/find_store.html#c')

# Wait two seconds and click the button "[주소]"
time.sleep(2)
driver.find_element(By.XPATH, '//*[@id="contentWrap"]/div[3]/div/div[1]/ul/li[2]/a').click()

C:\Users\kjaeh\AppData\Local\Temp\ipykernel_11328\283882113.py:1: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome('chromedriver.exe')


In [15]:
gu_names = list(starbucks['Area'].unique()) # Use the area information from 'starbucks'
gu_names

['강남구','강동구', '강북구', '강서구', '관악구', '광진구', '구로구', '금천구', '노원구', '도봉구', '동대문구', '동작구',
 '마포구', '서대문구', '서초구', '성동구', '성북구', '송파구', '양천구', '영등포구', '용산구', '은평구', '종로구', '중구',
 '중랑구'] 

['강남구',
 '강동구',
 '강북구',
 '강서구',
 '관악구',
 '광진구',
 '구로구',
 '금천구',
 '노원구',
 '도봉구',
 '동대문구',
 '동작구',
 '마포구',
 '서대문구',
 '서초구',
 '성동구',
 '성북구',
 '송파구',
 '양천구',
 '영등포구',
 '용산구',
 '은평구',
 '종로구',
 '중구',
 '중랑구']

In [16]:
# Variables

search_box = driver.find_element(By.XPATH, '//*[@id="keyword"]')
search_btn = driver.find_element(By.XPATH, '//*[@id="keyword_div"]/form/button')

In [17]:
# Function to get detailed (old) address from an overlay box

def GetDetailAddr(br):
    br.click()
    time.sleep(1)

    go_details = driver.find_element(By.CLASS_NAME, 'st_info_btn').find_element(By.TAG_NAME, 'img')
    go_details.click()
    time.sleep(1)

    addr_detail = driver.find_element(By.TAG_NAME, 'dd')
    target = addr_detail.text
    time.sleep(1)

    close = driver.find_element(By.CLASS_NAME, 'pop_close').find_element(By.TAG_NAME,'a')
    close.click()
    time.sleep(1)

    return target

In [18]:
# Function to generate empty dataframe

def GenerateDataFrame(Area = [], Branch = [], Addr = []):
    tmp = (
        {'Area': Area,
        'Branch' : Branch,
        'Address' : Addr,
        }
    )
    return pd.DataFrame(tmp)

ediya = GenerateDataFrame()
ediya.head()

Empty DataFrame
Columns: [Area, Branch, Address]
Index: []

In [19]:
for gu in tqdm_notebook(gu_names): 
    search_box.clear()
    search_box.send_keys('서울 ' + gu) # In case there are multiple cities having the same district name. (ex: 부산 강서구, 인천 중구)
    time.sleep(1)

    search_btn.click()
    time.sleep(1)

    places = driver.find_elements(By.CLASS_NAME, 'item')
    area = []; branch = []; address = []; 
    for p in places:
        tmp = p.text.split('\n')
        branch.append(tmp[0])

        if tmp[1][-1] == '동': 
            address.append(GetDetailAddr(p))
        else: 
            address.append(tmp[1])

        area = [gu for i in range(len(address))]

    ediya_tmp = GenerateDataFrame(area, branch, address)    
    ediya = pd.concat([ediya, ediya_tmp])

    time.sleep(1)

ediya.head()
# Some branches provide old address in an overlay box. Get information from it.

C:\Users\kjaeh\AppData\Local\Temp\ipykernel_11328\505152868.py:1: TqdmDeprecationWarning: This function will be removed in tqdm==5.0.0
Please use `tqdm.notebook.tqdm` instead of `tqdm.tqdm_notebook`
  for gu in tqdm_notebook(gu_names):


  0%|          | 0/25 [00:00<?, ?it/s]

Area       Branch                      Address
0  강남구      강남YMCA점             서울 강남구 논현동 230-2
1  강남구  강남구청역아이티웨딩점  서울 강남구 학동로 338 (논현동, 강남파라곤)
2  강남구      강남논현학동점      서울 강남구 논현로131길 28 (논현동)
3  강남구        강남대치점   서울 강남구 역삼로 415 (대치동, 성진빌딩)
4  강남구        강남도산점      서울 강남구 도산대로37길 20 (신사동)

In [20]:
ediya.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 717 entries, 0 to 26
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Area     717 non-null    object
 1   Branch   717 non-null    object
 2   Address  717 non-null    object
dtypes: object(3)
memory usage: 22.4+ KB


In [24]:
# Resetting the index
ediya.reset_index(inplace=True)

# Remove previous index
del ediya['index']

ediya.reset_index(inplace=True)
ediya.set_index('index')
ediya.info()

In [25]:
ediya.head()

index Area       Branch                      Address
0      0  강남구      강남YMCA점             서울 강남구 논현동 230-2
1      1  강남구  강남구청역아이티웨딩점  서울 강남구 학동로 338 (논현동, 강남파라곤)
2      2  강남구      강남논현학동점      서울 강남구 논현로131길 28 (논현동)
3      3  강남구        강남대치점   서울 강남구 역삼로 415 (대치동, 성진빌딩)
4      4  강남구        강남도산점      서울 강남구 도산대로37길 20 (신사동)

In [26]:
uniq = ediya['Area'].unique()
print(len(uniq), '\n', uniq)

25 
 ['강남구' '강동구' '강북구' '강서구' '관악구' '광진구' '구로구' '금천구' '노원구' '도봉구' '동대문구' '동작구'
 '마포구' '서대문구' '서초구' '성동구' '성북구' '송파구' '양천구' '영등포구' '용산구' '은평구' '종로구' '중구'
 '중랑구']


In [27]:
# ediya.to_csv('data/01_ediya.csv', sep =',', encoding= 'cp949')
driver.close()

## 문제 3. 스타벅스, 이디야 점포 정보를 토대로 한 데이터 분석

### 3 - (1) Information Gathering & Summary

<details>
<summary>기능 구현까지는 연결하지 못한 정보 수집 및 정리 과정을 담았습니다. (내용이 다소 깁니다)<br>채점에는 무관할 수 있겠지만, 읽어주시고 피드백 주신다면 정말 감사하겠습니다.<p></summary>
<div markdown="1">

> "이디야는 스타벅스의 임점 위치를 따라 개점하는, 이른바 '스타벅스 옆자리 전략'으로 무임승차한다."<br>


먼저 (기자(혹은 작성자)의 이름이 공개된) 온라인 기사와 블로그 포스팅 8건을 통해 현재까지의 주장 진행을 살피겠습니다. <br>
*(이러한 자료 선정 역시 데이터에 근거해서 할 수 있도록 성장하고 싶다는 생각이 듭니다.)*


위 주장을 참으로 소개하는 자료(이하 참-자료)는 총 5건, (부분) 거짓으로 소개하는 자료(이하 거짓-자료)는 총 3건입니다.


- 참-자료 5건에서는 "실제로 ... 발견하는 것이 어렵지 않다' (2015년), '실제로 ... 떠올리거나 살펴보시기 바랍니다' (2020년), ' 지금도 스타벅스 매장 주변을 보면 ..." (2022년) 등 생활권에서 육안으로 관찰되는 내용을 바탕으로 이디야의 입점 전략을 '미투 전략<sup>1</sup>'의 일환으로 설명합니다. 이 자료들 중에는 이디야커피 고위 관계자가 **"브랜드 론칭 초기에는 ...(중략)... 스타벅스 옆을 노리는 전략을 펼쳤다."**라고 시인한 내용도 있습니다. (2015년)


- 거짓-자료 3건에는 스타벅스와 이디야가 공개한 매장 데이터를 분석해 지도에 시각화한 자료를 토대로 스타벅스의 '허브 앤 스포크 전략<sup>2</sup>'과 이디야의 '포인트 투 포인트 전략<sup>3</sup>'을 비교합니다. 이 또한 **"사실 스타벅스는 옆은 ...(중략)... 이디야는 입점하기 힘들다"** (2019년)는 관계자의 귀띔이 소개됩니다. 


- 데이터를 근거로 주장하는 입장에서 봤을 때는, 참-자료의 갯수가 더 많더라도 이를 입증하는 시각화된 데이터가 없으니 데이터가 있는 거짓-자료의 손을 들어 "이디야는 스타벅스의 입점 위치를 따라하지 않는다"라고 자신있게 말하고 싶습니다. 하지만, 이디야가 론칭 초기에는 이른바 '스타벅스 옆자리 전략'을 펼쳤다는 고위 관계자의 인정이 있기 떄문에, 맨 처음 제기된 가설은 시점에 대한 보완과 함께 다음과 같이 수정되어야 할 필요가 있습니다.


> "이디야는 *(론칭 초기에는)* 이른바 '스타벅스 옆자리 전략'을 펼쳤지만, *(현재는)* **지방 및 특수 상권을 포함한 미개척 상권 개발**<sup>4</sup> 전략을 펼치고 있다."


물론 이 가설에도 (1) 론칭 초기와 현재를 구분하는 기준에 대한 근거, (2) <sup>4</sup>번 전략을 정의하는 기준 등 거쳐야 할 단계가 많겠지만 시간 문제, 데이터 접근 등 여러 측면에서 모두 진행하기는 어려웠습니다. 따라서 후술될 목표에 따라 시각화를 진행하겠습니다.



<sub><sup>1</sup>미투(Me-too) 전략: 성공한 브랜드를 모방하는 경영 전략<br>
<sup>2</sup>허브 앤 스포크(Hub and Spoke) 전략: 유동 인구가 많은 지역에 집중적으로 매장을 펼치는 전략<br>
<sup>3</sup>포인트 투 포인트(Point-to-point) 전략: 곳곳에 매장을 펼치는 전략<br>
<sup>4</sup>지방 및 특수 상권을 포함한 미개척 상권 개발: 이디야갸 공개한 입지전략</sub> 

**References**
1. 찬성-자료 <br>
- https://www.edaily.co.kr/news/read?newsId=01105366609365640 (이데일리, "이디야의 필승 전략..'스타벅스 옆을 사수하라'", 2015년) <br>
- http://www.iconsumer.or.kr/news/articleView.html?idxno=10133 (소비자평가, "‘스타벅스를 따라다닌 이디야’, 이디야 문창기 대표가 펼친 특별한 전략", 2019년) <br>
- https://m.blog.naver.com/PostView.naver?isHttpsRedirect=true&blogId=qaz030800&logNo=221680589189 (네이버 블로그[작성자: 윈에이드], "유명기업의 마케팅전략: '이디야(EDIYA)커피' 편, 2019년) <br>
- https://m.blog.naver.com/urimmoe/222027146026 (네이버 블로그[작성자: 로하스김], "커피전문점 이디야의 스타벅스 미투마케팅 전략", 2020년) <br>
- https://www.topdaily.kr/articles/90068 (톱데일리, "이디야커피 성장 비결, 스타벅스 따라다니기?", 2022년) <br>

2. 반대-자료 <br>
- https://www.seoul.co.kr/news/newsView.php?id=20160919004005 (서울신문, "스타벅스 vs 이디야, 매장 위치에 전략이 숨어 있다", 2016년) <br>
- https://brunch.co.kr/@tipoon1004/66 (브런치[작성자: 권오은], "이디야는 스타벅스 옆을 떠난 걸까", 2017년) <br>
- https://www.insight.co.kr/news/194890 (인사이트, "왜 스타벅스 옆에는 항상 토종 커피 브랜드 '이디야'가 있을까?", 2018년) <br>

</div>
</details>

### 3 - (3) Data Preprocessing

#### 1. Import Modules for Visualization

In [28]:
import pandas as pd
import numpy as np

import folium

import googlemaps
gmaps = googlemaps.Client(key = 'AIzaSyCqYCVCRygQxXhDzocSK30NneK9Bx8YGcA')

import json # Seoul District Border Lines
json_path = 'data/seoul_district.json'
json_load = json.load(open(json_path, encoding='utf-8'))


import time
from tqdm import tqdm_notebook

In [32]:
# starbucks = pd.read_csv('data/02_starbucks_geo.csv', encoding = 'cp949')
starbucks.head()

index Area    Branch                     Address
0      0  강남구     역삼이마트     서울특별시 강남구 역삼로 310 (역삼동)
1      1  강남구       삼성동  서울특별시 강남구 영동대로96길 12 (삼성동)
2      2  강남구      강남논현    서울특별시 강남구 강남대로 512 (논현동)
3      3  강남구  테헤란로아남타워    서울특별시 강남구 테헤란로 311 (역삼동)
4      4  강남구       삼성역    서울특별시 강남구 테헤란로 620 (대치동)

In [30]:
# ediya = pd.read_csv('data/02_ediya_geo.csv', encoding = 'cp949')
ediya.head()

index Area       Branch                      Address
0      0  강남구      강남YMCA점             서울 강남구 논현동 230-2
1      1  강남구  강남구청역아이티웨딩점  서울 강남구 학동로 338 (논현동, 강남파라곤)
2      2  강남구      강남논현학동점      서울 강남구 논현로131길 28 (논현동)
3      3  강남구        강남대치점   서울 강남구 역삼로 415 (대치동, 성진빌딩)
4      4  강남구        강남도산점      서울 강남구 도산대로37길 20 (신사동)

#### 2. Add Geometric Information - Functions

In [33]:
# Function: getting lat, lng info from dataframe's ['address'] column

def AddGeometryInfo(DataFrame):
    lat_list = []; lng_list = []
    for s in tqdm_notebook(DataFrame['Address']):
        try:
            g = gmaps.geocode(s)[0]
            lat = g['geometry']['location']['lat']
            lng = g['geometry']['location']['lng']
            lat_list.append(lat)
            lng_list.append(lng)
        except: # Some Address cannot get lat, lng. In that case, fill in the list with enpty str ''
            lat_list.append('')
            lng_list.append('')
            
    DataFrame['lat'] = lat_list
    DataFrame['lng'] = lng_list

    return DataFrame

In [34]:
# Function: return indexes which cannot call lat, lng information 

def getInavailAddr(DataFrame):
    idx = list(filter(lambda x: DataFrame['lat'][x] == '', range(len(DataFrame))))
    if len(idx) != 0:
        return idx
    else:
        print('No empty lat, lng')

In [35]:
# Fucnction: change the address into right one
    # return 4 words in the address (도로명주소 4번쨰 어절)

def AddrChange(DataFrame, idx):
    for i in idx:
        old = DataFrame['Address'][i]
        new = ' '.join(old.split()[:4]).strip(',') # strip for a single address with ','
        DataFrame.loc[i, 'Address'] = new

In [36]:
# Function: update lng, lat info of address changed.

def UpdateGeocode(DataFrame, idx):
    for i in idx:
        g = gmaps.geocode(DataFrame.loc[i, 'Address'])[0]
        lat = g['geometry']['location']['lat']
        lng = g['geometry']['location']['lng']
        
        DataFrame.loc[i, 'lat'] = lat
        DataFrame.loc[i, 'lng'] = lng

#### 3. Add Geometric Information - Perform

In [37]:
starbucks = AddGeometryInfo(starbucks)
starbucks.head()

C:\Users\kjaeh\AppData\Local\Temp\ipykernel_11328\588393876.py:5: TqdmDeprecationWarning: This function will be removed in tqdm==5.0.0
Please use `tqdm.notebook.tqdm` instead of `tqdm.tqdm_notebook`
  for s in tqdm_notebook(DataFrame['Address']):


  0%|          | 0/593 [00:00<?, ?it/s]

index Area    Branch                     Address        lat         lng
0      0  강남구     역삼이마트     서울특별시 강남구 역삼로 310 (역삼동)  37.499224  127.048424
1      1  강남구       삼성동  서울특별시 강남구 영동대로96길 12 (삼성동)  37.510701  127.063568
2      2  강남구      강남논현    서울특별시 강남구 강남대로 512 (논현동)   37.51353  127.031534
3      3  강남구  테헤란로아남타워    서울특별시 강남구 테헤란로 311 (역삼동)   37.50375  127.044836
4      4  강남구       삼성역    서울특별시 강남구 테헤란로 620 (대치동)   37.50916  127.065565

In [38]:
sb_addr = getInavailAddr(starbucks)
AddrChange(starbucks, sb_addr)
UpdateGeocode(starbucks, sb_addr)
getInavailAddr(starbucks)

No empty lat, lng


In [39]:
ediya = AddGeometryInfo(ediya)
ediya.head()

C:\Users\kjaeh\AppData\Local\Temp\ipykernel_11328\588393876.py:5: TqdmDeprecationWarning: This function will be removed in tqdm==5.0.0
Please use `tqdm.notebook.tqdm` instead of `tqdm.tqdm_notebook`
  for s in tqdm_notebook(DataFrame['Address']):


  0%|          | 0/717 [00:00<?, ?it/s]

index Area       Branch                      Address        lat         lng
0      0  강남구      강남YMCA점             서울 강남구 논현동 230-2  37.509902  127.035706
1      1  강남구  강남구청역아이티웨딩점  서울 강남구 학동로 338 (논현동, 강남파라곤)  37.516466  127.040419
2      2  강남구      강남논현학동점      서울 강남구 논현로131길 28 (논현동)  37.514227  127.028666
3      3  강남구        강남대치점   서울 강남구 역삼로 415 (대치동, 성진빌딩)  37.504296  127.063053
4      4  강남구        강남도산점      서울 강남구 도산대로37길 20 (신사동)  37.522241  127.031481

In [40]:
ed_addr = getInavailAddr(ediya)
AddrChange(ediya, ed_addr)
UpdateGeocode(ediya, ed_addr)
getInavailAddr(ediya)

No empty lat, lng


In [ ]:
# Save dataframes

# starbucks.to_csv('data/02_starbucks_geo.csv', sep= ',', encoding = 'cp949')
# ediya.to_csv('data/02_ediya_geo.csv', sep= ',', encoding = 'cp949') 


___

### 3 - (4) Data Visualization - Folium

#### 1. Functions & Variables

In [41]:
# Function: get lat, lng info from dataframe

def getLat(geocode):
    return geocode[0]['geometry']['location']['lat']

def getLng(geocode):
    return geocode[0]['geometry']['location']['lng']

In [42]:
# Function: generate a map
    # set "zoom_start = 13" to get a good view on each district

def GenerateMap(lat, lng):
    m = folium.Map(
        location = [lat, lng],
        zoom_start  = 13,
        tiles='CartoDB positron',
        width = 750,
        height = 750
    )
    return m

In [43]:
# Function: Add markers and cicles on the map

def EdiyaMarkers(DataFrame, Area, Map):
    df = DataFrame[DataFrame['Area'] == Area]
    for idx, rows in df.iterrows():            
        folium.Marker(
            [rows['lat'], rows['lng']],
            icon=folium.Icon(
                color='blue',
                icon='cloud'),
            tooltip = '이디야 ' + rows['Branch'], # Name '이디야'
            popup = folium.Popup(' '.join(rows['Address'].split(' ')[:3]),  max_width=1000),
        ).add_to(Map)
    
def StarbucksMarkers(DataFrame, Area, Map):
    df = DataFrame[DataFrame['Area'] == Area]
    for idx, rows in df.iterrows():            
        folium.Marker(
            [rows['lat'], rows['lng']],
            icon=folium.Icon(
                color='green',
                icon='star'),
            tooltip = '스타벅스 ' + rows['Branch'], # Name '스타벅스'
            popup = folium.Popup(' '.join(rows['Address'].split(' ')[:3]), max_width=1000),
        ).add_to(Map)

def StarbucksCircle(DataFrame, Area, Map):
    df = DataFrame[DataFrame['Area'] == Area]
    for idx, rows in df.iterrows():            
        folium.Circle(
            [rows['lat'], rows['lng']],
            radius=100,
            color = 'green', # 반경 100m 
            tooltip = '스타벅스 ' + rows['Branch'],
            popup = folium.Popup(' '.join(rows['Address'].split(' ')[:3]), max_width=1000),
        ).add_to(Map)

In [44]:
# Locations: Starting Points for each district

mid_area = {
'강남구': '서울특별시 강남구 남부순환로 2952', #대치역
'강동구': '서울특별시 강동구 양재대로 1572', #굽은다리역
'강북구': '서울특별시 강북구 수유동 338-46', #가오리역
'강서구': '서울특별시 강서구 공항대로 163', #마곡역
'관악구': '서울 관악구 신림동', #관악IC 
'광진구': '서울 광진구 자양로 167', #서울광진경찰서
'구로구': '서울특별시 구로구 경인로 40길 47', #개봉역 
'금천구': '서울 금천구 시흥대로73길 70', #금천구청 
'노원구': '서울특별시 노원구 상계로 182', #상계역 
'도봉구': '서울 도봉구 방학로2길 27', #방학1동주민센터
'동대문구': '서울 동대문구 왕산로 214', #청량리역
'동작구': '서울특별시 동작구 상도로 378', #숭실대입구역
'마포구': '서울특별시 마포구 월드컵로 77', #망원역
'서대문구': '서울 서대문구 연희로 248 서대문구청', #서대문구청
'서초구': '서울특별시 서초구 매헌로 116', #양재시민의숲역
'성동구': '서울특별시 성동구 고산자로 123', #응봉역 
'성북구': '서울 성북구 보문로 168 성북구청', #성북구청
'송파구': '서울특별시 송파구 송파대로 354', #송파구청
'양천구': '서울특별시 양천구 중앙로 261', #신정네거리역
'영등포구': '서울 영등포구 경인로 846', #영등포역 
'용산구': '서울 용산구 용산동4가 13', #용산공원 
'은평구': '서울특별시 은평구 서오릉로 63', #역촌역 
'종로구': '서울 종로구 삼청로11길 32-2', #삼청동청사
'중구': '서울특별시 중구 퇴계로 214', #충무로역
'중랑구': '서울 중랑구 망우로55길 11-10', #망우역
} 

#### 2. Generate maps & add markers

In [45]:
# variables - 25 maps in lists by district
maps = []; areas = list(mid_area.keys())

# Create a map 
for area in tqdm_notebook(mid_area.keys()):
    g = gmaps.geocode(mid_area[area])
    m = GenerateMap(getLat(g), getLng(g))
    
    StarbucksMarkers(starbucks, area, m)
    StarbucksCircle(starbucks, area, m)
    EdiyaMarkers(ediya, area, m)

    maps.append(m)

C:\Users\kjaeh\AppData\Local\Temp\ipykernel_11328\2918075724.py:5: TqdmDeprecationWarning: This function will be removed in tqdm==5.0.0
Please use `tqdm.notebook.tqdm` instead of `tqdm.tqdm_notebook`
  for area in tqdm_notebook(mid_area.keys()):


  0%|          | 0/25 [00:00<?, ?it/s]

In [46]:
# Check the map. Call the first and the last index map
maps[0] # Area == '강남구'

In [47]:
maps[24] # Area == '중랑구'

강남구부터 중랑구까지 총 25개 구에 대해 <br>
- 이디야와 스타벅스 각 지점명, 
- 이디야와 스타벅스 각 지점 '로/길' 단위까지의 주소 <br>

를 띄우는 marker를 표시한 지도를 완성했습니다. 


초록색 원(스타벅스 반경 100m)의 한 가운데 파란색 마커가 있는 경우가 있습니다. <br>
주소가 같아 순서 상 나중에 실행된 함수 EdiyaMarker()의 결광비니다. <br> 
첫 번째 목표인 '스타벅스 지점 반경 100m 내에 이디야 지점 위치'를 확인하기 위해 수정 없이 진행하겠습니다.<br>


___

### 3 - (5) Data Analysis

#### Method 1. Ediya branches in starbucks circle

스타벅스 지점의 folium.circle 내부에 이디야 지점의 marker가 위치해 있는지 육안으로 확인했습니다.
확인된 이디야 지점의 매장명과 구별 매장 수를 입력합니다.

> 강남구 = 역삼초고졈, 학동역점, 삼성중앙역점, 수서역점 (4) <br>
강동구 = 천호로데오점, 고덕점 (2) <br>
강북구 = 수유역점 (1) <br>
강서구 = 신방화역점, 마곡역점, 마곡파인스퀘어점, 가양역점, 강서nc점 (5)  <br>
관악구 = 낙성대역점, 관악구청점, 봉천역점 (3) <br>
광진구 = 뚝섬유원지역점, 건대스타시티점, 어린이대공원역점, 구의동점 (4) <br>
구로구 = 구로에이스점, 고척스카이돔점 (2) <br>
금천구 = 가산SKV1점, 가산그레이트점 (2) <br>
노원구 = 월계이마트점 (1)<br>
도봉구 = 없음 (0) <br>
동대문구 = 답십리역점, 경희대점, 외대점 (3) <br>
동작구 = 남성역점, 숭실대점, 노들점, 노량진역사점 (4) <br>
마포구 = 상암KBS점, 상암한샘점, 합정홈플러스점, 홍대중앙점, 서강대점 (5) <br>
서대문구 = 아현점, 이화여대점, 연세대점, 가재울중앙점 (4) <br>
서초구 = 방배백석점, 방배역점, 강남센트럴시티점, 잠원논현점, 신논현역점, 서초점, 서초역점 (7) <br>
성동구 = 성수중앙점, 뚝섬점, 왕십리비트플렉스점 (3) <br>
성북구 = 고대점, 한성대역점 (2)  <br>
송파구 = 가든파이브테크노관점, 가든파이브점, 송파하비오점, 올림픽공원점, 잠실나루역점(5) <br>
양천구 = 신월5동점, 목동파라곤점, 목동파리공원점 (3) <br>
영등포구 = 영등포knk점, 영등포점, 홈플러스점, kbs별관점, 여의도농협재단점, 국회대로점, 서여의도wja, 국회의사당역점 (8) <br>
용산구 = 순천향병원점, 숙대입구역점, (2) <br>
은평구 = 갈현점점, 구산역점, 녹번중앙점, 은평이마트점 (4) <br>
종로구 = 대학로점, 탑골공원점, 종각점, 삼일빌딩점, 종로타워점, 종로구청점, 광화문트윈트리타워점, 광화문점   (8) <br>
중구 = 황학롯데캐슬점, 동대문공원점, 충무필동점, 충무로점, 을지로3가점, IBK본점점, 명동중앙점, 회현점,   sk리더스점, 명동역점, 서울시청점, 태평로점 (12) <br>
중랑구 = 중랑역점 (1) <br>

In [48]:
# Contain numbers of overlapped branches
overlap = [4, 2, 1, 5, 3, 4, 2, 2, 1, 0, 3, 4, 5, 4, 7, 3, 2, 5, 3, 8, 2, 4, 8, 12, 1]

# Numbers of whole branches in each area
ediya_all = [len(ediya[ediya['Area'] == a]) for a in areas]

# Rate of overlapped branches over whole branches
rate = [round((o / e)* 100, 2) for o, e in zip(overlap, ediya_all)]

In [49]:
# Convert rate information to dataframe
df_rate = pd.DataFrame(
    {'Area': [a for a in areas],
    'Rate': [r for r in rate]}
)
df_rate.info()
'''25개 구에 대한 25개의 비율 정보 (float) 입력 확인'''

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25 entries, 0 to 24
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Area    25 non-null     object 
 1   Rate    25 non-null     float64
dtypes: float64(1), object(1)
memory usage: 528.0+ bytes


'25개 구에 대한 25개의 비율 정보 (float) 입력 확인'

In [50]:
df_rate.describe()

Rate
count  25.000000
mean   13.122000
std     8.451651
min     0.000000
25%     7.410000
50%    12.000000
75%    16.670000
max    40.000000

근접 입점 비율의 평균은 13.12%, 최대값은 40% (중구), 최소값은 0% (도봉구)입니다. <br>
목표값인 50%에 인접한 구는 하나도 없지만, 지역별 차이를 확인하기 위해 choropleth를 사용합니다. <br>
최대값인 40%에 가까울수록 초록색이 진하게 나오는 걸 확인할 수 있습니다.

In [51]:
seoul = folium.Map(
    location = [37.5502, 126.982], 
    height = 750, 
    width = 750,
    zoom_start=11, 
    tiles= 'CartoDB positron')

seoul.choropleth(
    geo_data = json_load,
    data = df_rate,
    columns = ['Area', 'Rate'],
    fill_color = 'BuGn',
    key_on = 'feature.id',
    fill_opacity = 0.7,
    line_opacity = 0.2
)
seoul

c:\Users\kjaeh\miniconda3\envs\DS_study\lib\site-packages\folium\folium.py:407: FutureWarning: The choropleth  method has been deprecated. Instead use the new Choropleth class, which has the same arguments. See the example notebook 'GeoJSON_and_choropleth' for how to do this.
  warnings.warn(


<결론 1> <br>
스타벅스 매장 100m 이내 이디야 매장이 입점한 비율, 인접 입점률이 50%를 넘기는 구가 없는 현재까지의 결과로는, 
**이디야가 '스타벅스 옆자리 전략'을 펼친다고 보기 어렵겠습니다.**

#### Method 2. Address and strategies

도로주소명의 구분 "대로 > 로 > (길, 번길)"에 따라 각 브랜드의 입점 비율을 분석합니다.

In [52]:
# Function - count ro and gil

def CntRoGil(DataFrame):
    list_ro = []; list_gil = []; list_dong = []
    for idx, row in DataFrame.iterrows():
        addr = row['Address']
        if addr.split(' ')[2][-1] == '로':
            list_ro.append(addr) 
        elif addr.split(' ')[2][-1] == '길':
            list_gil.append(addr)
        else:
            list_dong.append(addr)
    
    return [len(list_ro), len(list_gil), len(list_dong)]

In [53]:
# Functions - calculate the ratio

def CalRoGil(DataFrame):
    tmp = CntRoGil(DataFrame)

    hub_spoke = round(tmp[0] / sum(tmp) * 100, 2)
    point_point = round(tmp[1] / sum(tmp) * 100, 2)
    return [hub_spoke, point_point]

In [54]:
# Perform - Entire Seoul

sb_rat = CalRoGil(starbucks)
print(f'스타벅스 서울 전 지점 1. 대로/로 입점 비율: {sb_rat[0]}%, 2. 길/번길 입점 비율: {sb_rat[1]}%')

ed_rat = CalRoGil(ediya)
print(f'이디야 서울 전 지점 1. 대로/로 입점 비율: {ed_rat[0]}%, 2. 길/번길 입점 비율: {ed_rat[1]}%')

스타벅스 서울 전 지점 1. 대로/로 입점 비율: 83.81%, 2. 길/번길 입점 비율: 16.02%
이디야 서울 전 지점 1. 대로/로 입점 비율: 69.18%, 2. 길/번길 입점 비율: 27.89%


서울시 전체를 대상으로 한 두 브랜드의 주소별 대로/로 입점비율 차이는 14.63%를, 길/번길 입점비율 차이는 11.87%를 보입니다. 

In [56]:
# Perform - By Area

rate_dic = {}
for area in areas:
    tmp_sb = starbucks[starbucks['Area'] == area]
    tmp_ed = ediya[ediya['Area'] == area]
    sb_rat = CalRoGil(tmp_sb)
    ed_rat = CalRoGil(tmp_ed)

    rate_dic[area] = [sb_rat[0], ed_rat[0], sb_rat[1], ed_rat[1]]

In [57]:
df_rate['Starbucks Hub'] = [v[0] for v in list(rate_dic.values())]
df_rate['Ediya Hub'] = [v[1] for v in list(rate_dic.values())]
df_rate['Starbucks PtoP'] = [v[2] for v in list(rate_dic.values())]
df_rate['Ediya PtoP'] = [v[3] for v in list(rate_dic.values())]

for idx, row in df_rate.iterrows():
    df_rate.loc[idx, 'gap'] = abs(row['Starbucks Hub'] - row['Ediya Hub'])

df_rate.head()

Area   Rate  Starbucks Hub  Ediya Hub  Starbucks PtoP  Ediya PtoP    gap
0  강남구   9.09          82.02      40.91           17.98       52.27  41.11
1  강동구   7.41          88.24      77.78           11.76       22.22  10.46
2  강북구   5.00         100.00      85.00            0.00       15.00  15.00
3  강서구  11.11          92.00      75.56            8.00       24.44  16.44
4  관악구  10.71          91.67      85.71            8.33       14.29   5.96

In [59]:
print(f'이디야 대로/로 입점 비율이 50%를 넘는 지역 수: {len(df_rate[df_rate["Ediya Hub"] > 50])}')
df_rate[df_rate['Ediya Hub'] > 50][df_rate['gap'] < 5]

이디야 대로/로 입점 비율이 50%를 넘는 지역 수: 23


C:\Users\kjaeh\AppData\Local\Temp\ipykernel_11328\3541682819.py:2: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  df_rate[df_rate['Ediya Hub'] > 50][df_rate['gap'] < 5]


Area   Rate  Starbucks Hub  Ediya Hub  Starbucks PtoP  Ediya PtoP   gap
11  동작구  18.18          81.82      77.27           18.18       22.73  4.55
22  종로구  25.81          57.50      58.06           42.50       25.81  0.56

총 25개의 구에서 이디야의 대로/로 입점 비율이 50%이상인 곳은 23개 구이며, (강남구, 도봉구 제외)<br>
그 중에서 스타벅스의 대로/로 입점비율과의 차이가 5% 미만인 곳은 2개 구 입니다. (동작구, 종로구)

In [60]:
# 이디야 입점전략 확인 - 포인트 투 포인트

seoul2 = folium.Map(
    location = [37.5502, 126.982], 
    height = 750, 
    width = 750,
    zoom_start=11, 
    tiles= 'CartoDB positron')

seoul2.choropleth(
    geo_data = json_load,
    data = df_rate,
    columns = ['Area', 'Ediya PtoP'],
    fill_color = 'PuBu',
    key_on = 'feature.id',
    fill_opacity = 0.7,
    line_opacity = 0.2
)
seoul2

c:\Users\kjaeh\miniconda3\envs\DS_study\lib\site-packages\folium\folium.py:407: FutureWarning: The choropleth  method has been deprecated. Instead use the new Choropleth class, which has the same arguments. See the example notebook 'GeoJSON_and_choropleth' for how to do this.
  warnings.warn(


이디야의 길/번길 입점 비율 (포인트 투 포인트 비율) 이 높을수록, 지도에 파란색이 강해집니다. <br>
지도에서 파란색이 연해진 부분은 대로/로 입점 비율(허 앤 스포크 비율)이 비교적 높은 곳입니다.

In [61]:
# 스타벅스 입점전략 확인 - 포인트 투 포인트

seoul3 = folium.Map(
    location = [37.5502, 126.982], 
    height = 750, 
    width = 750,
    zoom_start=11, 
    tiles= 'CartoDB positron')

seoul3.choropleth(
    geo_data = json_load,
    data = df_rate,
    columns = ['Area', 'Starbucks Hub'],
    fill_color = 'BuGn',
    key_on = 'feature.id',
    fill_opacity = 0.7,
    line_opacity = 0.2
)
seoul3

스타벅스의 대로/로 입점 비율 (허브 앤 스포크 비율)이 높을 수록 지도의 초록색이 더 강해집니다. <br>

<결론 2> <br>
스타벅스는 유동인구가 많은 대로변에 직영점을 오픈하는 '허브 앤 스포크' 전략, <br>
이디야는 구석마다 가맹점을 오픈하는 '포인트 투 포인트' 전략을 사용합니다. <br>

시각화된 두 지도 seoul2와 seoul3를 통해, 
스타벅스의 허브 앤 스포크 비율이 높은 곳과, 이디야의 포인트 투 포인트 비율이 높은 위치가 상이함을 확인할 수 있습니다. <br>
이는 스타벅스와 이디야의 입점 전략이 다름을 의미하며, 이를 통해 '스타벅스 옆자리 전략' 주장의 설득력은 한층 더 약해집니다.

___

감사합니다.